sentry-core/src/test/java/io/sentry/core/transport/HttpTransportTest.kt

```diff
@@ -86,7 +86,7 @@ class HttpTransportTest {
     fun `uses Retry-After header if X-Sentry-Rate-Limit is not set when sending an event`() {
         val transport = fixture.getSUT()
 
-        whenever(fixture.connection.inputStream).thenThrow(IOException())
+        whenever(fixture.serializer.serialize(any<SentryEvent>(), any())).thenThrow(IOException())
```

Comment:
It's a good sign for your tests when you change so much in the actual class and so little in your tests 👍

Reply:
`l`: We need to change the exact same code in roughly 10 places in this test. That is a sign for me, that we should extract this functionality into a function for example.

Reply:
I extracted to a method, I do understand the duplication is not good, but it helps to read and understand the test when it's all there in the same block of code, otherwise to understand a test you need to navigate thru all the methods, but indeed, in this case, it's worth it.
I tried to use `inline/reified` but mockito is complaining, I'll check this out later.

Reply:
If you name the method properly you shouldn't need to read it's content to understand what it is doing.